io/src/main/scala/sbt/internal/io/TimeSource.scala

```diff
@@ -0,0 +1,47 @@
+package sbt.internal.io
+import scala.concurrent.duration._
+
+private[io] trait TimeSource {
```

Comment:
Could you provide documentation on why this is needed, especially as implicit?

Reply:
Sure. I got lazy because it was private[io]. The commit message in https://github.com/sbt/io/pull/226/commits/f7ecd379d470ac8f2251ce039cdb20d5243889dc sort of explains it, but I'll add scaladoc. The short of it is that using Deadline.now leads to flaky tests.

Reply:
https://github.com/sbt/io/compare/b86688d63868d82d264c10f6160ac8e29439d375..7e2770102359c6688868627f124c3306649c1f63